common/views/components/styled/Space.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import styled from 'styled-components';
+import type { ComponentType } from 'react';
+// This component will output vertical space from one of five pre-defined sizes: xs, s, m, l, and xl.
+// Each of these sizes has a pixel value for each of the three breakpoints, 'small', 'medium', and 'large'.
+// The component expects a `size` prop, and by default will use `margin-bottom` to add space.
+// This can be overriden to include one or more of `margin-top`, `margin-bottom`, `padding-top`, `padding-bottom`, `top`, and `bottom`.
+
+// <Space size='s' /> -- outputs a small amount of (default) margin-bottom
+// <Space size='xl' properties=`{['padding-top', 'padding-bottom']}` /> -- outputs extra-large padding-top and padding-bottom
+
+type SpaceSize = 'xs' | 's' | 'm' | 'l' | 'xl';
+type VerticalSpaceProperty =
+  | 'margin-top'
+  | 'margin-bottom'
+  | 'padding-top'
+  | 'padding-bottom'
+  | 'top'
+  | 'bottom';
+
+type HorizontalSpaceProperty =
+  | 'margin-left'
+  | 'margin-right'
+  | 'padding-left'
+  | 'padding-right'
+  | 'left'
+  | 'right';
```

Comment:
Could the combinations fo these be further solidified?
e.g.. you could not have padding-top/xl padding-bottom/s

Reply:
You can only add one `size` for each of the `h` and `v` props, so what you're saying is already true (but maybe I'm missing something?).

Reply:
```JSX
<Component />
<SpacingComponentThatWorksOutSpacingDependantOnSiblings />
<Component />
```

☝️ would that ever be a thing we would like to work towards?

I'm not sure what the conditions would be e.g. What about the adjacent components would allow you to infer the spacing.

Reply:
I've thought about this a bit, but I was running up against the problem you mention – what is it about the adjacent siblings that should govern the amount of space?

I think we should keep thinking about it. It could possibly be something that we deal with by the concept of e.g. 'layouts' with different zones, and those zones have the requisite space applied, rather than the components themselves.